test/enhance-request.test.js

```diff
@@ -29,27 +33,31 @@ test('Should enhance the Node.js core request/response objects', t => {
   fastify.listen(0, (err, address) => {
     t.error(err)
     sget({
-      method: 'GET',
-      url: address
+      method: 'POST',
+      url: `${address}?foo=bar`,
+      body: { bar: 'foo' },
+      json: true
     }, (err, res, data) => {
       t.error(err)
     })
   })
 })
 
 test('Should not enhance the Node.js core request/response objects when there are no middlewares', t => {
-  t.plan(9)
+  t.plan(11)
   const fastify = Fastify()
   t.teardown(fastify.close)
 
-  fastify.register(middiePlugin)
+  fastify.register(middiePlugin, { hook: 'preHandler' })
 
-  fastify.get('/', async (req, reply) => {
+  fastify.post('/', async (req, reply) => {
     t.equal(req.raw.originalUrl, undefined)
     t.equal(req.raw.id, undefined)
     t.equal(req.raw.hostname, undefined)
     t.equal(req.raw.ip, undefined)
     t.equal(req.raw.ips, undefined)
+    t.same(req.raw.body, undefined)
+    t.same(req.raw.query, undefined)
```

Comment:
should this _not_ be `undefined` based on the changes in this PR?

Reply:
This section is for when theres `this[kMiddlewares].length > 0`, my change follows the others in that theyre not set when no middlewares are set